src/sentry/api/endpoints/organization_tagkey_values.py

```diff
@@ -31,6 +32,12 @@ def get(self, request: Request, organization, key) -> Response:
 
         sentry_sdk.set_tag("query.tag_key", key)
 
+        if dataset := request.GET.get("dataset"):
```

Comment:
Downside of this is that `dataset` starts as a string and becomes type `Dataset` later on which breaks typing a bit